homeassistant/components/light/hive.py

```diff
@@ -90,11 +99,24 @@ def turn_on(self, **kwargs):
         if ATTR_COLOR_TEMP in kwargs:
             tmp_new_color_temp = kwargs.get(ATTR_COLOR_TEMP)
             new_color_temp = round(1000000 / tmp_new_color_temp)
+        if ATTR_RGB_COLOR in kwargs:
+            get_new_color = kwargs.get(ATTR_RGB_COLOR)
+            tmp_new_color = colorsys.rgb_to_hsv(get_new_color[0],
+                                                get_new_color[1],
+                                                get_new_color[2])
+            hue = int(round(tmp_new_color[0] * 360))
+            saturation = int(round(tmp_new_color[1] * 100))
+            value = int(round((tmp_new_color[2] / 255) * 100))
+            new_color = (hue, saturation, value)
 
         if new_brightness is not None:
             self.session.light.set_brightness(self.node_id, new_brightness)
         elif new_color_temp is not None:
-            self.session.light.set_colour_temp(self.node_id, new_color_temp)
+            self.session.light.set_color_temp(self.node_id,
+                                              self.light_device_type,
+                                              new_color_temp)
+        elif new_color is not None:

```

Comment:
Is there a reason on the device side why these checks are exclusive? More than one of these corresponding kwargs might be passed to the method at the same time, so it would be good to support setting all with one call of the method.

Reply:
Hi @MartinHjelmare 

There are 3 different hive bulbs

1) Standard bulb which just supports brightness
2) warmtocool bulb which support brightness and  colour temp
3) colourbulb which supports the same as point 2 plus colour 

Because there is so many different combinations of changing the lights we decided to base it on what was being received out of HA. Also there is an added pain that the colourbulb and warmtocool bulb gets the min/max colour temp the same way but sets it in 2 different ways.

Reply:
It also mimics the way that the hive app updates one thing at a time as well

Reply:
I think these three cases of bulbs are pretty much standard in the field.

The standard way to support this method in home assistant is to do all of the state changes from one call of `turn_on`. So if you decide to not support that, it needs to be clearly documented.

If you can I would recommend you to support changes all types of states in one go.